x11-libs/wxGTK: Filter flags to fix build issue

EmRowlands opened the pull request:
Filtering these FLAGS fix a build issue that occurs when creating pre-compiled headers (search logs for 'bk-make-pch'). This bug can alternatively be fixed by adding the --disable-precomp-headers configure option.

Closes: https://bugs.gentoo.org/504204
Signed-off-by: Emily Rowlands <emily@erowl.net>

As I said in the commit message, this could also be fixed by disabling precompiled headers. I wasn't sure which solution would be best, so I just chose one. If disabling PCHs would be the better option, I can change the fix to use that instead.

thesamesam commented:
Could we flip to foricng off PCH instead?

We've decided it's not worth the hassle:
```
commit 8c4774042b7fdfb08e525d8af4b7912f26a2fdce
Author: Sam James <sam@gentoo.org>
Date:   Wed Jan 12 02:04:15 2022 +0000

    profiles/base: [QA] unmask PCH for current GCCs to avoid rebuilds

    There's no need to force people to rebuild their current
    GCC just to turn off PCH.

    Adding this because of the previous change adding pch to use.mask.

    Signed-off-by: Sam James <sam@gentoo.org>

commit fb809aeadee57ffa24591e60cfb41aecd4823090
Author: Sam James <sam@gentoo.org>
Date:   Wed Jan 12 01:16:55 2022 +0000

    profiles/base: [QA] mask USE=pch

    PCH is notoriously fragile & unstable. It often leads
    to (unclear) build failures and has questionable
    value in terms of performance, at least for the general
    case.

    Users are free to unmask it if they wish at their
    own risk.

    Bug: https://bugs.gentoo.org/753323
    Bug: https://bugs.gentoo.org/822690
    Signed-off-by: Sam James <sam@gentoo.org>
```

EmRowlands commented:
I've gated `--disable-precomp-headers` behind `USE=pch`, rather than disabling it completely, similar to app-emulation/virtualbox and others.

thesamesam commented on x11-libs/wxGTK/wxGTK-3.0.4-r3.ebuild:
```diff
@@ -94,6 +94,11 @@ multilib_src_configure() {
+		# PCHes are unstable and are disabled in-tree where possible
+		# See bug #504204
+		# Commits 8c4774042b7fdfb08e525d8af4b7912f26a2fdce, fb809aeadee57ffa24591e60cfb41aecd4823090
+		$(use_enable pch precomp-headers)
```
Wrong place

EmRowlands replied:
Whoops, misapplied my own patch

Fixed in ed9c73deaedebd147462b51bab58e858277bfea0

thesamesam commented on x11-libs/wxGTK/wxGTK-3.0.4-r302.ebuild:
```diff
@@ -85,6 +85,11 @@ src_prepare() {
 		configure || die
+		# PCHes are unstable and are disabled in-tree where possible
+		# See bug #504204
+		# Commits 8c4774042b7fdfb08e525d8af4b7912f26a2fdce, fb809aeadee57ffa24591e60cfb41aecd4823090
```
Wrong place

EmRowlands replied:
Fixed in ed9c73deaedebd147462b51bab58e858277bfea0

EmRowlands commented:
I've just been investigating why the build produces a bunch of QA errors from -Waddress, but these appear before my change as well.

If you're happy with all the changes, feel free to merge when ready.

thesamesam commented:
Happy to start seeing these PRs! I'll ask leio to take a look but to me, it's an obvious win and a strong argument would be needed to grant wxGTK an exception from the PCH policy.

EmRowlands commented:
It looks like @leio hasn't committed since March. Is there someone else who can review this?

SoapGentoo reviewed (approved):
LGTM, please rebase, since I removed the unused `3.0.4-r302` and `3.0.4-r303` versions

EmRowlands commented:
I've rebased onto master, and replaced the deprecated dependency on `virtual/jpeg`. It seems like a trivial change, so I thought I'd add it on the same PR.
